### ISSUE TITLE
Explanation of rules

### ISSUE THREAD

**Issue by fkromer** [git-afsantos/haros_plugins#7]
Are you planing to add `rules.xml` files with explanations of the rules [like this one for the cppcheck_plugin](https://github.com/git-afsantos/haros_plugins/blob/advanced/cppcheck_plugin/rules.xml) for the other plugins as well? If yes let me know. I can help out with PRs.

**EDIT:** It seems like you are using the .xml file for the analyzer (elements `<pattern> ... </pattern>`). I am not sure if PR implementations are straightforward then...

**Comment by git-afsantos** [comment]
Indeed, that XML file is a feature of `cppcheck`, so it is not straightforward to replicate it to other plugins. There are some workarounds, however. The simplest one is to modify the rules in `plugin.yaml` to have a `doc` field with a long documentation string that explains the rule, or how it is implemented. Or `description` could be used for this, and add a `summary` for the short strings used right now in `description`.

Perhaps a more structured form of documentation can be justified, but I am on a pretty busy period right now. If all goes well with my current milestone, in the future some rules might incorporate something similar to those `<pattern>` in XML that will be automated.
